src/Kiota.Builder/Refiners/RubyRefiner.cs

```diff
@@ -35,10 +40,64 @@ public override void Refine(CodeNamespace generatedCode)
                 var declaration = currentClass.StartBlock as CodeClass.Declaration;
                 declaration.Implements.Add(new CodeType(currentClass) {
                     IsExternal = true,
-                    Name = $"Parsable",
+                    Name = $"MicrosoftKiotaAbstractions::Parsable",
                 });
             }
             CrawlTree(currentElement, AddParsableInheritanceForModelClasses);
         }
+        protected static void AddInheritedAndMethodTypesImports(CodeElement currentElement) {
+            if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model)) {
+                var declaration = currentClass.StartBlock as CodeClass.Declaration;
+                if(declaration.Inherits != null){
+                    currentClass.AddUsing(new CodeUsing(currentElement) { Name = declaration.Inherits.Name, Declaration = declaration.Inherits});
+                }
+            }
+            CrawlTree(currentElement, (x) => AddInheritedAndMethodTypesImports(x));
+        }
+
+        private static void AddGetterAndSetterMethodsRuby(CodeElement current, HashSet<CodePropertyKind> propertyKindsToAddAccessors, bool removeProperty, bool parameterAsOptional) {
+            if(!(propertyKindsToAddAccessors?.Any() ?? true)) return;
```

Comment:
this looks very similar to the method available in the common language refiner, I was curious to why you needed a different method?